File: Medium/238. Product of Array Except Self.py
class Solution(object):
    def productExceptSelf(self, nums):
        ans = [1]

        for i in range(1, len(nums)):
            ans.append(nums[i - 1] * ans[i - 1])

        j = len(nums) - 1
        X = 1

        while(j >= 0):
            ans[j] = ans[j] * X
            X *= nums[j]
            j-=1

        return ans
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        

File: js/238. Product of Array Except Self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    let answer = [1];

    for (var i = 1; i < nums.length; i++) {
        answer[i] = nums[i - 1] * answer[i - 1];
    }

    let R = 1;

    for (var i = nums.length - 1; i >= 0; i--) {
        answer[i] = answer[i] * R;
        R *= nums[i];
    }

    return answer;
};
